modules/users/client/services/authentication.client.service.js

```diff
@@ -1,12 +1,81 @@
-'use strict';
+(function() {
+  'use strict';
 
-// Authentication service for user variables
-angular.module('users').factory('Authentication', ['$window',
-  function ($window) {
-    var auth = {
-      user: $window.user
+  angular
+    .module('users')
+    .service('Authentication', Authentication);
+
+  Authentication.$inject = ['$q', 'localStorageService', '$resource', '$http', '$location', '$state'];
+  function Authentication($q, localStorageService, $resource, $http, $location, $state) {
+
+    var readyPromise = $q.defer();
+
+    var service = {
+      ready: readyPromise.promise,
+      user: null,
+      token: null,
+      login: login,
+      signout: signout,
+      refresh: refresh
     };
 
-    return auth;
+    function login(user, token) {
+      setUser(user);
+      setToken(token);
+      setHeader();
+      readyPromise.resolve(service);
+    }
+
+    function setUser(user) {
+      service.user = user;
+    }
+
+    function setToken(token) {
+      service.token = token;
+      localStorageService.set('token', token);
```

Comment:
Well for one it fallsback to cookies if localstorage is not supported.  After writing the tests, I had the same thoughts.  Especially for CSRF you don't want to store in cookies.  Would be willing to remove the localstorage module and just use the browser API if we don't want to support older browsers.  Let me know what you think.